mozilla_schema_generator/generic_ping.py

```diff
@@ -101,8 +101,11 @@ def make_schemas(env: Schema, probes: List[Probe], config: Config,
         schemas.append(final_schema)
         for s in schemas:
             for key in config.get_match_keys():
-                s.delete_group_from_schema(key + ("propertyNames",), propagate=False)
-                s.delete_group_from_schema(key + ("additionalProperties",), propagate=True)
+                if s.get(key + ("propertyNames",)) is not None:
+                    s.delete_group_from_schema(key + ("propertyNames",), propagate=False)
+
+                if s.get(key + ("additionalProperties",)) is not None:
```

Comment:
> Can you describe why the behavior of _get needs to change?

Some schemas don't have these fields. Previously, `_get` just returned and exception

Reply:
Okay, I see. I'd rather add a flag to `_get` to let it behave differently for these cases, and leave everything else as-is.

Reply:
I wrapped these 2 cases into `try` ... `except` which might be a little more elegant than adding an extra flag

Reply:
That sounds like the right thing to do, thanks!